### PR TITLE
All-in-one commit to complete Pull Request #56

### LINKED ISSUE #42: Strings into a separated file
Take all the strings of the app, and put them together in a `Localizable.strings` file.

Then create an enum (or struct with constants) to use them.

```swift
struct LocalizedString {

  static let example = "example".localized()
  // ...

}
```
(This is an example where `example` is the string id in the `Localizable.strings` file)

### LINKED ISSUE #43: Replace `Centre` occurrences by `Lieu`
After #42, we can then easily replace occurrences of `Centre` by `Lieu` in the `Localizable.strings` file.

Linked to https://github.com/CovidTrackerFr/vitemadose-front/issues/18 (follow what was done on frontend and android)

### LINKED ISSUE #57: App language is set to English on the App Store
Not sure if it is the right place to tell you this but FYI the app language is set to English on the App Store product page. 

Capture d’écran 2021-04-22 à 15 26 02

### PR DESCRIPTION
**Pull request by PlugNPush**
Fix #42 & Fix #43 & Fix #57

### CODE REVIEW

File: ViteMaDose/Helpers/Extensions/String+DateParsing.swift [hunk 1]
```diff
@@ -19,4 +19,17 @@ extension String {
         let date = toDate(nil, region: region) ?? toISODate(nil, region: region)
         return date?.toString(style)
     }
+
+    // Localization
```
Comment (NathanFallet) [inline ViteMaDose/Helpers/Extensions/String+DateParsing.swift:23]: In the case you add it here, rename the file `String+Common(s).swift`

File: ViteMaDose/Views/CentresList/Cells/CentresTitleCell.swift [hunk 2]
```diff
@@ -36,15 +31,15 @@ extension CentresTitleCell {
 
         guard dosesCount > 0 else {
             let title = NSMutableAttributedString(
-                string: "\(Constant.titleNoDoseText) \(countyName)",
+                string: "\(LocalizedString.VaccinationLocationsList.noResults) \(countyName)",
```
Comment (NathanFallet) [inline ViteMaDose/Views/CentresList/Cells/CentresTitleCell.swift:34]: Not better using a formatted string?

File: ViteMaDose/Views/CentresList/Cells/CentresTitleCell.swift [hunk 3]
```diff
@@ -36,15 +31,15 @@ extension CentresTitleCell {
...
                 attributes: attributes
             )
             title.setColorForText(textForAttribute: countyName, withColor: .mandy)
             return title
         }
 
-        let dosesCountString = "\(String(dosesCount)) \(Constant.titleSecondPartText)"
-        let titleString = "\(Constant.titleFirstPartText) \(dosesCountString) \(Constant.titleThirdText) \(countyName)"
+        let dosesCountString = dosesCount > 1 ? LocalizedString.VaccinationLocationsList.doses(dosesCount) : LocalizedString.VaccinationLocationsList.dose(dosesCount)
```
Comment (NathanFallet) [inline ViteMaDose/Views/CentresList/Cells/CentresTitleCell.swift:41]: Same thing for formatted string here

**Review by NathanFallet — CHANGES_REQUESTED** [review]

File: ViteMaDose/Views/CentresList/Cells/CentresTitleCell.swift (hunk 3 above)
Comment (PlugNPush) [reply ViteMaDose/Views/CentresList/Cells/CentresTitleCell.swift:41]: Alternative way: I could create a new function and make the dose/doses localized strings private, and the single function will decide accordingly what to return. What do you think about that?

**Review by NathanFallet — APPROVED** [review]

File: ViteMaDose/Helpers/Utils/LocalizedString.swift [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    enum GenericError {
+
+        static let cancelButton = "generic_error.cancel_button".localized()
+        static let title = "generic_error.title".localized()
+        static let retryButton = "generic_error.retry_button".localized()
+        static let defaultMessage = "generic_error.default_message".localized()
+
+    }
+
+    fileprivate enum PluralHandler {
```
Comment (NathanFallet) [inline ViteMaDose/Helpers/Utils/LocalizedString.swift:117]: @victor-sarda Just need you to see if this is okay for you. If it is, it's mergeable.

**Review by NathanFallet — APPROVED** [review]

File: ViteMaDose/Helpers/Utils/LocalizedString.swift (hunk 4 above)
Comment (PlugNPush) [reply ViteMaDose/Helpers/Utils/LocalizedString.swift:117]: In any case, this PR will not be merged into develop but in a dedicated branch, making a lot of modifications possible before the final PR.

Comment (NathanFallet) [reply ViteMaDose/Helpers/Utils/LocalizedString.swift:117]: If he wants he can push a commit to your branch directly before merging it.
Also he told me he was making a branch with your changes, adding some cleanup

**Comment by victor-sarda** [comment]
Closing (relates to #73 )

**Pull request closed without merging**